### DISCUSSION TITLE
【V5 API】v5版本连线上设置多个图标，如何维持多个图标间的间距不会随着节点连线长度的变化而发生变化？

### DISCUSSION THREAD

**Discussion by ghost** [antvis/G6#6506] [Q&A]
<img width="440" alt="image" src="https://github.com/user-attachments/assets/a2f6ed50-baf4-4182-85c8-b92a956f7764">

如第一张图所示的一个连接关系图，在A-B的连线上，存在三个图标，表示一些连线的属性或者状态。

<img width="466" alt="image" src="https://github.com/user-attachments/assets/96cacbfa-708f-4d07-b55f-5bf25d992b2e">

在v5 api里，目前了解到，三个图标需要分别绘制，每个图标的位置都是单独计算的，这就会引发这样一个情况，随着A-B之间的连线长度变大或者缩小，这三个图标之间的间距也会变大，影响最终呈现效果，如第二张图所示。

<img width="466" alt="image" src="https://github.com/user-attachments/assets/f1812f4e-18ce-46e6-8754-ba230b837f19">

期望效果如第三张图，我想到的是可以将这三个图标归到一个分组内，通过定位分组，定位三个图标在连线上的位置，也可以通过分组，固定三个图标的位置关系，不会随着连线长度的变化而发生变化，影响呈现效果。

在v5版本中有什么方式可以实现这样的效果吗？

**Comment by Rey-Wang** [comment]
我觉得有两个个思路，你看行不行
- 直接用字体 icon
- 自定节点，把这几个图片放在一个 group 中 render

**Reply by ghost** [reply]
字体icon不太方便，还是倾向于图片这些，方便用户自己设置导入。我想的也是第二种方案，通过自定义节点，放到一个group中，但我不知道该怎么添加group，像在V4 API中，有addGroup这样的API可以设置，在V5里面应该如何添加group。

另外还有一个问题是关于自定义线上增加图标定位的问题，在V4的API中有getPoint(0.5)这样的API，V5该如何使用？

**Reply by Rey-Wang** [reply]
V4没用过，我是正在学 V5 怎么用，按照我目前的了解，自定义节点继承 Line 类，重写render方法，你可以看看这个实例
https://g6.antv.vision/zh/examples/element/custom-edge/#extra-label

通过 upsert 方法新增一个 Group 的 DisplayObject上去，坐标根据线的中心点算一下

**Comment by Aarebecca** [comment]
需要看你的图标是如何放置的，正如 @Rey-Wang 提到的，你可以把几个图标置于同一个 group 中

**Reply by ghost** [reply]
@Aarebecca 这个我也想过，参考v4版本addGroup这样的API，但是在v5版本里面，文档中没有相关Group的描述了，upsert方法增加Group，Group的样式支持哪些？是否是按照BaseNode样式来设置。

另外还有个疑问，目前在API中，好像也没有了类似v4版本的getPoint这样的API，应该如何在连线上定位，有案例给些参考么？
